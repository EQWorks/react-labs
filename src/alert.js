import React from 'react'
import PropTypes from 'prop-types'
import MUIAlert from '@material-ui/lab/Alert'
import AlertTitle from '@material-ui/lab/AlertTitle'

const Alert = ({ severity, message, header, width, ...props }) => {
  const dimensions = { height: 'auto', width }

  return (
    <div style={dimensions}>
      {header ? (
        <MUIAlert severity={severity} {...props}>
          <AlertTitle>
            <strong>{header}</strong>
          </AlertTitle>
          {message}
        </MUIAlert>
      ) : (
        <MUIAlert severity={severity} {...props}>
          {message}
        </MUIAlert>
      )}
    </div>
  )
}

Alert.propTypes = {
  /**
    * The header of the component.
  */
  header: PropTypes.string,
  /**
    * The message of the component.
  */
  message: PropTypes.string.isRequired,
  /**
    * The severity of the alert. This defines the color and icon used.
  */
  severity: PropTypes.oneOf(['error', 'warning', 'info', 'success']).isRequired,
  /**
    * The variant to use.
  */
  variant: PropTypes.oneOf(['standard', 'outlined', 'filled']).isRequired,
  /**
    * The width of the component.
  */
  width: PropTypes.string,
}

Alert.defaultProps = {
  variant: 'standard',
  width: '100%',
}

export default Alert
