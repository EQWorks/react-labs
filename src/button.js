import React from 'react'
import PropTypes from 'prop-types'
import Button from '@material-ui/core/Button'

const ButtonComponent = ({ children, type, ...props }) => {
  const styleProps = {
    color: 'primary',
    type: 'primary'
  }
  if (type === 'tertiary') {
    props.variant = 'text'
  } else if (type === 'secondary') {
    props.variant = 'outlined'
  } else {
    props.variant = 'contained'
  }

  return <Button {...styleProps} {...props}>{children}</Button>
}

ButtonComponent.propTypes = {
  children: PropTypes.node,
  type: PropTypes.oneOf(['primary', 'secondary', 'tertiary'])
}

ButtonComponent.defaultProps = {
  children: 'Button',
  type: 'primary'
}

export default ButtonComponent