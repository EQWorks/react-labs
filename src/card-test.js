import React from "react";
import PropTypes from "prop-types";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";

const CardComponent = ({ type }) => {
  return (
    <Card className={type}>
      <CardContent>
        <Typography>Hello</Typography>
      </CardContent>
    </Card>
  );
};

CardComponent.propTypes = {
  type: PropTypes.oneOf(["primary", "secondary", "success"]),
};

CardComponent.defaultProps = {
  type: "primary",
};

export default CardComponent;
