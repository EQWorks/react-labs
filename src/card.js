import React from 'react'
import PropTypes from 'prop-types'
import { makeStyles } from '@material-ui/core/styles'
import MUICard from '@material-ui/core/Card'
import CardActions from '@material-ui/core/CardActions'
import CardHeader from '@material-ui/core/CardHeader'
import CardContent from '@material-ui/core/CardContent'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'

import { StyledCardContainer } from '.'

const useStyles = makeStyles((theme) => {
  return {
    sm: {
      maxWidth: 200,
      height: 220,
      padding: theme.spacing(1),
      position: 'relative',
    },
    md: {
      maxWidth: 500,
      height: 270,
      padding: theme.spacing(1),
      position: 'relative',
    },
    actions: {
      position: 'absolute',
      bottom: 0,
      right: 0,
      left: 0,
      padding: theme.spacing(1.2),
    },
  }
})

const Card = ({
  cardTitle,
  alignTitle,
  variantTitle,
  cardContent,
  alignContent,
  variantContent,
  cardAction,
  size,
  headerProps,
  actionSide,
  width,
  height,
  styledContainerProps,
}) => {
  const classes = useStyles()
  const dimensions = { maxWidth: width, height }
  const alignTxt = (placement) => {
    if (placement === 'start') return 'left'
    if (placement === 'end') return 'right'
    return placement
  }
  const cardBody = (
    <>
      {Object.entries(headerProps).length > 0 && (
        <CardHeader {...headerProps} />
      )}
      <CardContent>
        <Grid
          container
          item
          xs={12}
          justify={alignTitle === 'center' ? 'center' : `flex-${alignTitle}`}
        >
          <Typography
            variant={variantTitle}
            component="span"
            align={alignTxt(alignTitle)}
          >
            {cardTitle}
          </Typography>
        </Grid>
        <Grid
          container
          item
          xs={12}
          justify={
            alignContent === 'center' ? 'center' : `flex-${alignContent}`
          }
        >
          <Typography
            variant={variantContent}
            component="span"
            align={alignTxt(alignContent)}
          >
            {cardContent}
          </Typography>
        </Grid>
      </CardContent>
      {cardAction && (
        <Grid
          className={classes.actions}
          container
          alignItems="flex-end"
          justify={actionSide === 'center' ? 'center' : `flex-${actionSide}`}
        >
          <CardActions disableSpacing>{cardAction}</CardActions>
        </Grid>
      )}
    </>
  )
  if (Object.entries(styledContainerProps).length > 0)
    return (
      <StyledCardContainer {...styledContainerProps}>
        {cardBody}
      </StyledCardContainer>
    )

  return (
    <MUICard data-testid='card' style={dimensions} className={classes[size]}>
      {cardBody}
    </MUICard>
  )
}

Card.propTypes = {
  /**
    * The vertical alignment of the action item.
  */
  actionSide: PropTypes.oneOf(['start', 'center', 'end']),
  /**
    * The vertical alignment of the title.
  */
  alignTitle: PropTypes.oneOf(['start', 'center', 'end']),
  /**
    * The vertical alignment of the content.
  */
  alignContent: PropTypes.string,
  /**
    * The action item of the component.
  */
  cardAction: PropTypes.any,
  /**
    * The content of the component.
  */
  cardContent: PropTypes.any.isRequired,
  /**
    * The title of the component.
  */
  cardTitle: PropTypes.string,
  /**
    * The properties of the component header.
  */
  headerProps: PropTypes.object,
  /**
    * The height of the component.
  */
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  /**
    * The size of the component.
  */
  size: PropTypes.oneOf(['sm', 'md']),
  /**
    * The properties of the component.
  */
  styledContainerProps: PropTypes.object,
  /**
    * The theme typography styles to the component content.
  */
  variantContent: PropTypes.oneOf([
    'h1',
    'h2',
    'h3',
    'h4',
    'h5',
    'h6',
    'subtitle1',
    'subtitle2',
    'body1',
    'body2',
    'caption',
    'button',
    'overline',
    'srOnly',
    'inherit',
  ]),
  /**
    * The theme typography styles to the component title.
  */
  variantTitle: PropTypes.oneOf([
    'h1',
    'h2',
    'h3',
    'h4',
    'h5',
    'h6',
    'subtitle1',
    'subtitle2',
    'body1',
    'body2',
    'caption',
    'button',
    'overline',
    'srOnly',
    'inherit',
  ]),
  /**
    * The width of the component.
  */
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
}

Card.defaultProps = {
  actionSide: 'start',
  alignContent: 'start',
  alignTitle: 'start',
  headerProps: {},
  size: 'md',
  styledContainerProps: {},
  variantContent: 'body2',
  variantTitle: 'h5',
}

export default Card
