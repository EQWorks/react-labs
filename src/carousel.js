import React, { useState, useRef } from "react";
import StyledCardContainer from "../src/styled-card-container";
import {
  Avatar,
  Chip,
  CardContent,
  Grid,
  Link,
  Typography,
  makeStyles,
  IconButton,
} from "@material-ui/core";
import CheckCircleOutlineRoundedIcon from "@material-ui/icons/CheckCircleOutlineRounded";
import { data3 } from "../stories/data/cardInfo";
import DynamicButton from "./dynamic-button";
import { typography } from "@storybook/theming";
import Slider from "react-slick";
import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";
import "./css/carousel.css";
import ArrowBackIosRoundedIcon from "@material-ui/icons/ArrowBackIosRounded";
import ArrowForwardIosRoundedIcon from "@material-ui/icons/ArrowForwardIosRounded";

const useStyles = makeStyles((theme) => ({
  sliderControl: {
    padding: "0px",
    marginLeft: theme.spacing(2),
  },
  textarea: {
    display: "flex",
    justifyContent: "space-between",
    marginBottom: theme.spacing(1),
  },
  wrapper: {
    display: "flex",
    flexWrap: "nowrap",
    overflowX: "auto",
    "&::-webkit-scrollbar": {
      display: "none",
    },
  },
  link: {
    display: "flex",
    alignItems: "flex-end",
    paddingRight: theme.spacing(2),
  },
  container: {
    width: "50px",
  },
  header: {
    display: "flex",
    justifyContent: "space-between",
  },
  content: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "space-between",
    height: "240px",
  },
  tag: {
    display: "flex",
    justifyContent: "flex-end",
  },
  chip: {
    borderRadius: "4px",
    margin: theme.spacing(0.5),
  },
}));

const Carousel = () => {
  const classes = useStyles();
  const customSlider = useRef();
  const settings = {
    dots: true,
    infinite: true,
    speed: 500,
    slidesToShow: 3,
    slidesToScroll: 3,
  };


  const gotoPrev = () => {
    customSlider.current.slickPrev()
  }

  const gotoNext = () => {
    customSlider.current.slickNext()
  }

  const topSection = (
    <Grid container className={classes.textarea}>
      <Grid item>
        <Typography variant="subtitle1">Bundles</Typography>
        <Typography variant="body1">
          Bundles are the collection of segments and layers packaged based on a
          specific persona.
        </Typography>
      </Grid>
      <Grid item className={classes.link}>
        <Link href="#" variant="body1" color="primary">
          View all
        </Link>
        <IconButton
          classes={{ root: classes.sliderControl }}
          onClick={() => gotoPrev()}
        >
          <ArrowBackIosRoundedIcon fontSize="small" />
        </IconButton>
        <IconButton
          classes={{ root: classes.sliderControl }}
          onClick={() => gotoNext()}
        >
          <ArrowForwardIosRoundedIcon fontSize="small" />
        </IconButton>
      </Grid>
    </Grid>
  );

  const contents = ({ name, description, type, price, category }) => (
    <React.Fragment>
      <Grid item>
        <Grid item className={classes.tag}>
          <Chip className={classes.chip} size="small" label={type} />
          <Chip className={classes.chip} size="small" label={category} />
        </Grid>
      </Grid>
      <Grid item>
        <Grid item className={classes.header}>
          <Typography variant="subtitle1">{name}</Typography>
        </Grid>
        <Typography variant="body1" gutterBottom>
          {description}
        </Typography>
        <Typography variant="h6">{price}/mo</Typography>
      </Grid>
    </React.Fragment>
  );

  return (
    <div>
      {topSection}
      <Slider {...settings} className={classes.slider} ref={customSlider}>
        {data3.map((cardInfo, i) => (
          <StyledCardContainer
            pattern={{
              style: 3,
              image: cardInfo.image,
            }}
          >
            <CardContent className={classes.content}>
              {contents(cardInfo)}
            </CardContent>
          </StyledCardContainer>
        ))}
      </Slider>
    </div>
  );
};

export default Carousel;
