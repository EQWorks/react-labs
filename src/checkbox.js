import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { makeStyles } from '@material-ui/core/styles'
import MUICheckbox from '@material-ui/core/Checkbox'
import FormControlLabel from '@material-ui/core/FormControlLabel'

const CHECKMARK =
  'url("data:image/svg+xml;charset=utf-8,%3Csvg xmlns=\'http://www.w3.org/2000/svg\' viewBox=\'0 0 16 16\'%3E%3Cpath' +
  ' fill-rule=\'evenodd\' clip-rule=\'evenodd\' d=\'M12 5c-.28 0-.53.11-.71.29L7 9.59l-2.29-2.3a1.003 ' +
  '1.003 0 00-1.42 1.42l3 3c.18.18.43.29.71.29s.53-.11.71-.29l5-5A1.003 1.003 0 0012 5z\' fill=\'%23fff\'/%3E%3C/svg%3E")'

const useStyles = makeStyles((theme) => {
  return {
    root: {
      margin: theme.spacing(0.5),
      '&&:hover': {
        backgroundColor: 'transparent',
      },
    },
    icon: {
      backgroundColor: theme.palette.grey[50],
      border: `1px solid ${theme.palette.grey[400]}`,
      borderRadius: 3,
      height: 16,
      width: 16,
      'input:disabled ~ &': {
        opacity: 0.5,
        pointerEvents: 'none',
      },
      'input:focus ~ &': {
        outline: `1px solid ${theme.palette.primary.main}`,
      },
      'input:hover ~ &': {
        backgroundColor: theme.palette.action.hover,
        border: `1px solid ${theme.palette.primary[100]}`,
        color: theme.palette.action.hover,
        transition: 'all .2s',
      },
    },
    checkedIcon: {
      backgroundColor: theme.palette.primary.main,
      backgroundImage: CHECKMARK,
      border: '1px solid rgba(0,0,0,0)',
      borderRadius: 3,
      height: 16,
      width: 16,
      'input:disabled ~ &': {
        opacity: 0.5,
      },
      'input:hover ~ &': {
        backgroundColor: theme.palette.primary[800],
        border: `1px solid ${theme.palette.primary[800]}`,
        transition: 'all .2s',
      },
    },
  }
})

const Checkbox = ({ checked, disabled, id, label }) => {
  const classes = useStyles()
  const [checkedState, setCheckedState] = useState(checked)
  const checkOnChange = (e) => {
    setCheckedState(e.target.checked)
  }
  return (
    <FormControlLabel
      control={
        <MUICheckbox
          disabled={disabled}
          disableRipple
          checked={checkedState}
          checkedIcon={
            <span className={checkedState ? classes.checkedIcon : classes.icon} />
          }
          className={classes.root}
          icon={<span className={classes.icon} />}
          id={id}
          onChange={checkOnChange}
        />
      }
      data-testid='checkbox'
      htmlFor={id}
      label={label}
    />
  )
}

Checkbox.propTypes = {
  /**
    * If `true`, the component is checked.
  */
  checked: PropTypes.bool,
  /**
    * If `true`, the checkbox will be disabled.
  */
  disabled: PropTypes.bool,
  /**
    * The event source of the callback. You can pull out the new checked state by accessing `event.target.checked` (boolean).
  */
  onChange: PropTypes.func,
  /**
    * The id of the element.
  */
  id: PropTypes.string,
  /**
    * The text to be used in an enclosing label element.
  */
  label: PropTypes.string,
}

Checkbox.defaultProps = {
  checked: false,
  disabled: false,
}

export default Checkbox
