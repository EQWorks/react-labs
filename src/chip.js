import React from 'react'
import PropTypes from 'prop-types'
import { makeStyles, useTheme } from '@material-ui/core/styles'
import { fade } from '@material-ui/core/styles/colorManipulator'
import MUIChip from '@material-ui/core/Chip'
import clsx from 'clsx'

const useStyles = makeStyles((theme) => ({
  chip: {
    backgroundColor: (styleProps) => theme.palette[styleProps.backgroundColor].main,
    color: (styleProps) => styleProps.color,
    margin: theme.spacing(1),
  },
  clickable: {
    '&:hover, &:focus': {
      backgroundColor: (styleProps) => fade(theme.palette[styleProps.backgroundColor].main, 0.75),
    },
  },
  onDelete: {
    '&:hover, &:focus': {
      backgroundColor: (styleProps) => `${fade(theme.palette[styleProps.backgroundColor].main, 0.15)} !important`,
    },
  },
  outlined: {
    backgroundColor: `${theme.palette.common.white} !important`,
    border: (styleProps) => `1px solid ${theme.palette[styleProps.backgroundColor].main}`,
    color: (styleProps) => `${theme.palette[styleProps.backgroundColor].main} !important`,
  },
  outlinedClickable: {
    '&:hover, &:focus': {
      backgroundColor: (styleProps) => `${fade(theme.palette[styleProps.backgroundColor].main, 0.15)} !important`,
    },
  },
  rectangle: {
    borderRadius: '4px',
  },
}))

const Chip = ({ clickable, color, onDelete, rectangle, variant, ...props }) => {
  const theme = useTheme()
  const styleProps = {
    backgroundColor: color,
    color: (['primary', 'success', 'warning'].includes(color))
      ? theme.palette.common.black : theme.palette.common.white,
  }
  const classes = useStyles(styleProps)

  return <MUIChip
    className={clsx({
      [classes.chip]: true,
      [classes.clickable]: clickable,
      [classes.onDelete]: onDelete,
      [classes.outlined]: (variant === 'outlined'),
      [classes.outlinedClickable]: (variant === 'outlined' && clickable),
      [classes.rectangle]: rectangle,
    })}
    clickable={clickable}
    data-testid='chip'
    onDelete={onDelete}
    variant={variant}
    {...props}
  />
}

Chip.propTypes = {
  /**
    * If true, the chip will appear clickable.
  */
  clickable: PropTypes.bool,
  /**
    * The color of the element.
  */
  color: PropTypes.oneOf(['primary', 'secondary', 'success', 'error', 'warning', 'info']).isRequired,
  /**
    * Callback function fired when the delete icon is clicked. If set, the delete icon will be shown.
  */
  onDelete: PropTypes.func,
  /**
    * If true, the element shape will be rectangular.
  */
  rectangle: PropTypes.bool,
  /**
    * The variant to use.
  */
  variant: PropTypes.oneOf(['default', 'outlined']),
}

Chip.defaultProps = {
  clickable: false,
  color: 'primary',
  rectangle: false,
  variant: 'outlined',
}

export default Chip
