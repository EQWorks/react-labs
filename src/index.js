export { default as WidgetTrend } from './widget-trend' // deprecated
export { default as WidgetNumber } from './widget-number' // deprecated
export { default as WidgetStats } from './widget-stats'
export { default as DataTable } from './data-table'
export { default as QuickFilters } from './quick-filters'
export { default as DynamicButton } from './dynamic-button'
export { default as DynamicDataTable } from './dynamic-data-table'
export { default as Table } from './table'
