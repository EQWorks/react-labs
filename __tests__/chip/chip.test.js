import React from 'react'
import { render } from '@testing-library/react'

import { Chip } from '../../src/index'

describe('render', () => {
  it('should render ok with minimal props passed', () => {
    const { getByTestId } = render(<Chip label='My Chip' />)
    const chip = getByTestId('chip')
    // snapshot
    expect(chip).toMatchSnapshot()
    // text content matches
    // expect(chip.textContent).toBe('my chip')
    // not loading
    // expect(chip).toHaveStyle('color: #FFFFFF')
    // not disabled
    // expect(chip).not.toBeDisabled()
    // medium size
    // expect(chip).not.toHaveClass('MuiButton-sizeSmall', 'MuiButton-sizeLarge')
    // primary type
    // expect(chip).toHaveAttribute('type', 'primary')
  })
})
