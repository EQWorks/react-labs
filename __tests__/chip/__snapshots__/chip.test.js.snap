// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should render ok with minimal props passed 1`] = `
<div
  class="MuiChip-root makeStyles-chip-1 makeStyles-chip-7 makeStyles-outlined-4 makeStyles-outlined-10 MuiChip-outlined"
  data-testid="chip"
>
  <span
    class="MuiChip-label"
  >
    My Chip
  </span>
</div>
`;
